internal/ingress/controller/kong.go

```diff
@@ -391,7 +391,7 @@ func (n *NGINXController) syncServices(ingressCfg *ingress.Configuration) (bool,
 							outOfSync = true
 						}
 
-						if kongIngress.Proxy.Retries > 0 &&
+						if kongIngress.Proxy.Retries >= 0 &&
```

Comment:
The default value for kongIngress.Proxy.Retries will be 0, meaning that this branch will evaluate to true if there a KongIngress resource associated with an Ingress. 
This will change the behavior as it will set the retries to 0.

The reason is, we can't differentiate between an explicit user setting it to 0 and an absent value.
This can be solved by using pointers instead of plain `int`.